crates/swc/tests/fixture/sourcemap/004/output/index.map

```diff
@@ -1,5 +1,5 @@
 {
-  "mappings": "AEACA,CAAAA,KAAK,gBAAmB,GAAGA,KAAK,gBAAmB,IAAI,EAAE,AAAD,EAAGC,IAAI,CAAC;IAC7D;QAAC;KAAI;IACL;QACU,MAAY,SACdC,CAAuB,EACvBC,CAAmB,EACnBC,CAAmB,EACrB;YACE;gBFNXC,IAAA,SAAAC,CAAA,EAAA;gBAAA,IAAAC,IAAAD,EAAAC,IAAA;gBAAA,OAAA,CAAA,GCDDC,EAAAC,GAAA,EAAA,OAAA;oBACAC,UAAAH,EAAAI,GAAA;gBACA;YACA;YAKYP,EAAoBQ,CAAC,CAACT,IACDC,EAAoBS,CAAC,CAACV,GAAqB;gBACvCW,SAAS,WAAY;oBACtC,OAAqBA;gBDZjB;gBAAqBC,SAAA,WAAA;oBAC3C,OAAOV;gBAAM;YACd;YAAA,IAAAG,IAAAJ,EAAA,OAAAU,IAAA,CAAA;QCKM;QAKP,MAAA,SAAAZ,CAAA,EAAAc,CAAA,EAAAZ,CAAA,EAAA;YAAAa,CAAAA,OAAAC,QAAA,GAAAD,OAAAC,QAAA,IAAA,EAAA,EAAAjB,IAAA,CAAA;gBAAA;gBAAA,WAAA;oBAAA,OAAAG,EAAA;gBAAA;aAAA;QAAA;IAAA;IAAA,SAAAA,CAAA,EAAA;QAAAA,EAAAe,CAAA,CAAA,GAAA;YAAA;YAAA;YAAA;SAAA,EAAA,WAAA;YAAA,OAAAf,EAAAA,EAAAgB,CAAA,GAAA;QAAA,IAAAC,OAAAjB,EAAAe,CAAA;IAAA;CAAA",
+  "mappings": ";;;KACS",
```

Comment:
This worries me, but it seems to be the input sourcemap is really bad:

- [input](https://evanw.github.io/source-map-visualization/#MzY2MwAoc2VsZlsid2VicGFja0NodW5rX05fRSJdID0gc2VsZlsid2VicGFja0NodW5rX05fRSJdIHx8IFtdKS5wdXNoKFsKICAgIFsxNThdLAogICAgewogICAgICAgIC8qKiovIDI5NDM6IC8qKiovIGZ1bmN0aW9uICgKICAgICAgICAgICAgX191bnVzZWRfd2VicGFja19tb2R1bGUsCiAgICAgICAgICAgIF9fd2VicGFja19leHBvcnRzX18sCiAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX18KICAgICAgICApIHsKICAgICAgICAgICAgInVzZSBzdHJpY3QiOwogICAgICAgICAgICBfX3dlYnBhY2tfcmVxdWlyZV9fLnIoX193ZWJwYWNrX2V4cG9ydHNfXyk7CiAgICAgICAgICAgIC8qIGhhcm1vbnkgZXhwb3J0ICovIF9fd2VicGFja19yZXF1aXJlX18uZChfX3dlYnBhY2tfZXhwb3J0c19fLCB7CiAgICAgICAgICAgICAgICAvKiBoYXJtb255IGV4cG9ydCAqLyBfX05fU1NHOiBmdW5jdGlvbiAoKSB7CiAgICAgICAgICAgICAgICAgICAgcmV0dXJuIC8qIGJpbmRpbmcgKi8gX19OX1NTRzsKICAgICAgICAgICAgICAgIH0sCiAgICAgICAgICAgICAgICAvKiBoYXJtb255IGV4cG9ydCAqLyBkZWZhdWx0OiBmdW5jdGlvbiAoKSB7CiAgICAgICAgICAgICAgICAgICAgcmV0dXJuIC8qIGJpbmRpbmcgKi8gU3RhdGljUGFnZTsKICAgICAgICAgICAgICAgIH0sCiAgICAgICAgICAgICAgICAvKiBoYXJtb255IGV4cG9ydCAqLwogICAgICAgICAgICB9KTsKICAgICAgICAgICAgLyogaGFybW9ueSBpbXBvcnQgKi8gdmFyIHJlYWN0X2pzeF9ydW50aW1lX19XRUJQQUNLX0lNUE9SVEVEX01PRFVMRV8wX18gPQogICAgICAgICAgICAgICAgX193ZWJwYWNrX3JlcXVpcmVfXyg0NTEyKTsKCiAgICAgICAgICAgIHZhciBfX05fU1NHID0gdHJ1ZTsKICAgICAgICAgICAgZnVuY3Rpb24gU3RhdGljUGFnZShfcmVmKSB7CiAgICAgICAgICAgICAgICB2YXIgZGF0YSA9IF9yZWYuZGF0YTsKICAgICAgICAgICAgICAgIHJldHVybiAvKiNfX1BVUkVfXyovICgwLAogICAgICAgICAgICAgICAgcmVhY3RfanN4X3J1bnRpbWVfX1dFQlBBQ0tfSU1QT1JURURfTU9EVUxFXzBfXy5qc3gpKCJkaXYiLCB7CiAgICAgICAgICAgICAgICAgICAgY2hpbGRyZW46IGRhdGEuZm9vLAogICAgICAgICAgICAgICAgfSk7CiAgICAgICAgICAgIH0KCiAgICAgICAgICAgIC8qKiovCiAgICAgICAgfSwKCiAgICAgICAgLyoqKi8gNzEzOTogLyoqKi8gZnVuY3Rpb24gKAogICAgICAgICAgICBfX3VudXNlZF93ZWJwYWNrX21vZHVsZSwKICAgICAgICAgICAgX191bnVzZWRfd2VicGFja19leHBvcnRzLAogICAgICAgICAgICBfX3dlYnBhY2tfcmVxdWlyZV9fCiAgICAgICAgKSB7CiAgICAgICAgICAgICh3aW5kb3cuX19ORVhUX1AgPSB3aW5kb3cuX19ORVhUX1AgfHwgW10pLnB1c2goWwogICAgICAgICAgICAgICAgIi9zdGF0aWMiLAogICAgICAgICAgICAgICAgZnVuY3Rpb24gKCkgewogICAgICAgICAgICAgICAgICAgIHJldHVybiBfX3dlYnBhY2tfcmVxdWlyZV9fKDI5NDMpOwogICAgICAgICAgICAgICAgfSwKICAgICAgICAgICAgXSk7CiAgICAgICAgICAgIGlmIChmYWxzZSkgewogICAgICAgICAgICB9CgogICAgICAgICAgICAvKioqLwogICAgICAgIH0sCiAgICB9LAogICAgLyoqKioqKi8gZnVuY3Rpb24gKF9fd2VicGFja19yZXF1aXJlX18pIHsKICAgICAgICAvLyB3ZWJwYWNrUnVudGltZU1vZHVsZXMKICAgICAgICAvKioqKioqLyB2YXIgX193ZWJwYWNrX2V4ZWNfXyA9IGZ1bmN0aW9uIChtb2R1bGVJZCkgewogICAgICAgICAgICByZXR1cm4gX193ZWJwYWNrX3JlcXVpcmVfXygoX193ZWJwYWNrX3JlcXVpcmVfXy5zID0gbW9kdWxlSWQpKTsKICAgICAgICB9OwogICAgICAgIC8qKioqKiovIF9fd2VicGFja19yZXF1aXJlX18uTygwLCBbNzc0LCA4ODgsIDE3OV0sIGZ1bmN0aW9uICgpIHsKICAgICAgICAgICAgcmV0dXJuIF9fd2VicGFja19leGVjX18oNzEzOSk7CiAgICAgICAgfSk7CiAgICAgICAgLyoqKioqKi8gdmFyIF9fd2VicGFja19leHBvcnRzX18gPSBfX3dlYnBhY2tfcmVxdWlyZV9fLk8oKTsKICAgICAgICAvKioqKioqLyBfTl9FID0gX193ZWJwYWNrX2V4cG9ydHNfXzsKICAgICAgICAvKioqKioqLwogICAgfSwKXSk7CgovLyMgc291cmNlTWFwcGluZ1VSTD1kYXRhOmFwcGxpY2F0aW9uL2pzb247YmFzZTY0LGV3b2dJQ0FnSW5abGNuTnBiMjRpT2lBekxBb2dJQ0FnSW1acGJHVWlPaUFpZUNJc0NpQWdJQ0FpYldGd2NHbHVaM01pT2lBaU96czdPenM3T3pzN096czdPenRCUVVGbExGTkJRVk5CTEZWQlFWUXNUMEZCT0VJN1FVRkJRU3hOUVVGU1F5eEpRVUZSTEZGQlFWSkJMRWxCUVZFN1FVRkRNME1zYzBKQlFVODdRVUZCUVN4alFVRk5RU3hKUVVGSkxFTkJRVU5ETzBGQlFWZ3NTVUZCVUR0QlFVTkVPenM3T3pzN096dEJRMFJFTzBGQlEwRTdRVUZEUVR0QlFVTkJMR1ZCUVdVc2JVSkJRVThzUTBGQlF5eEpRVUU0UWp0QlFVTnlSRHRCUVVOQk8wRkJRMEVzVDBGQlR5eExRVUZWTEVWQlFVVXNSVUZKWkR0QlFVTk1JaXdLSUNBZ0lDSnpiM1Z5WTJWeklqb2dXd29nSUNBZ0lDQWdJQ0ozWldKd1lXTnJPaTh2TGk5d1lXZGxjeTl6ZEdGMGFXTXVhbk1pTEFvZ0lDQWdJQ0FnSUNKM1pXSndZV05yT2k4dkxpNHZMaTR2TGk0dmNHRmphMkZuWlhNdmJtVjRkQzlrYVhOMEwySjFhV3hrTDNkbFluQmhZMnN2Ykc5aFpHVnljeTl1WlhoMExXTnNhV1Z1ZEMxd1lXZGxjeTFzYjJGa1pYSXVhbk0vY0dGblpUMGxNa1p6ZEdGMGFXTW1ZV0p6YjJ4MWRHVlFZV2RsVUdGMGFEMXdjbWwyWVhSbExXNWxlSFF0Y0dGblpYTWxNa1p6ZEdGMGFXTXVhbk1oSWdvZ0lDQWdYU3dLSUNBZ0lDSnpiM1Z5WTJWelEyOXVkR1Z1ZENJNklGc0tJQ0FnSUNBZ0lDQWlaWGh3YjNKMElHUmxabUYxYkhRZ1puVnVZM1JwYjI0Z1UzUmhkR2xqVUdGblpTaDdJR1JoZEdFZ2ZTa2dlMXh1SUNCeVpYUjFjbTRnUEdScGRqNTdaR0YwWVM1bWIyOTlQQzlrYVhZK1hHNTlYRzVjYm1WNGNHOXlkQ0JoYzNsdVl5Qm1kVzVqZEdsdmJpQm5aWFJUZEdGMGFXTlFjbTl3Y3lncElIdGNiaUFnY21WMGRYSnVJSHRjYmlBZ0lDQndjbTl3Y3pvZ2UxeHVJQ0FnSUNBZ1pHRjBZVG9nZTF4dUlDQWdJQ0FnSUNCbWIyODZJQ2RpWVhJbkxGeHVJQ0FnSUNBZ2ZTeGNiaUFnSUNCOUxGeHVJQ0I5WEc1OVhHNGlMQW9nSUNBZ0lDQWdJQ0pjYmlBZ0lDQW9kMmx1Wkc5M0xsOWZUa1ZZVkY5UUlEMGdkMmx1Wkc5M0xsOWZUa1ZZVkY5UUlIeDhJRnRkS1M1d2RYTm9LRnRjYmlBZ0lDQWdJRndpTDNOMFlYUnBZMXdpTEZ4dUlDQWdJQ0FnWm5WdVkzUnBiMjRnS0NrZ2UxeHVJQ0FnSUNBZ0lDQnlaWFIxY200Z2NtVnhkV2x5WlNoY0luQnlhWFpoZEdVdGJtVjRkQzF3WVdkbGN5OXpkR0YwYVdNdWFuTmNJaWs3WEc0Z0lDQWdJQ0I5WEc0Z0lDQWdYU2s3WEc0Z0lDQWdhV1lvYlc5a2RXeGxMbWh2ZENrZ2UxeHVJQ0FnSUNBZ2JXOWtkV3hsTG1odmRDNWthWE53YjNObEtHWjFibU4wYVc5dUlDZ3BJSHRjYmlBZ0lDQWdJQ0FnZDJsdVpHOTNMbDlmVGtWWVZGOVFMbkIxYzJnb1cxd2lMM04wWVhScFkxd2lYU2xjYmlBZ0lDQWdJSDBwTzF4dUlDQWdJSDFjYmlBZ0lnb2dJQ0FnWFN3S0lDQWdJQ0p1WVcxbGN5STZJRnNLSUNBZ0lDQWdJQ0FpVTNSaGRHbGpVR0ZuWlNJc0NpQWdJQ0FnSUNBZ0ltUmhkR0VpTEFvZ0lDQWdJQ0FnSUNKbWIyOGlDaUFnSUNCZENuMEsxMTI4AHsKICAgICJ2ZXJzaW9uIjogMywKICAgICJmaWxlIjogIngiLAogICAgIm1hcHBpbmdzIjogIjs7Ozs7Ozs7Ozs7Ozs7QUFBZSxTQUFTQSxVQUFULE9BQThCO0FBQUEsTUFBUkMsSUFBUSxRQUFSQSxJQUFRO0FBQzNDLHNCQUFPO0FBQUEsY0FBTUEsSUFBSSxDQUFDQztBQUFYLElBQVA7QUFDRDs7Ozs7Ozs7QUNERDtBQUNBO0FBQ0E7QUFDQSxlQUFlLG1CQUFPLENBQUMsSUFBOEI7QUFDckQ7QUFDQTtBQUNBLE9BQU8sS0FBVSxFQUFFLEVBSWQ7QUFDTCIsCiAgICAic291cmNlcyI6IFsKICAgICAgICAid2VicGFjazovLy4vcGFnZXMvc3RhdGljLmpzIiwKICAgICAgICAid2VicGFjazovLy4uLy4uLy4uL3BhY2thZ2VzL25leHQvZGlzdC9idWlsZC93ZWJwYWNrL2xvYWRlcnMvbmV4dC1jbGllbnQtcGFnZXMtbG9hZGVyLmpzP3BhZ2U9JTJGc3RhdGljJmFic29sdXRlUGFnZVBhdGg9cHJpdmF0ZS1uZXh0LXBhZ2VzJTJGc3RhdGljLmpzISIKICAgIF0sCiAgICAic291cmNlc0NvbnRlbnQiOiBbCiAgICAgICAgImV4cG9ydCBkZWZhdWx0IGZ1bmN0aW9uIFN0YXRpY1BhZ2UoeyBkYXRhIH0pIHtcbiAgcmV0dXJuIDxkaXY+e2RhdGEuZm9vfTwvZGl2PlxufVxuXG5leHBvcnQgYXN5bmMgZnVuY3Rpb24gZ2V0U3RhdGljUHJvcHMoKSB7XG4gIHJldHVybiB7XG4gICAgcHJvcHM6IHtcbiAgICAgIGRhdGE6IHtcbiAgICAgICAgZm9vOiAnYmFyJyxcbiAgICAgIH0sXG4gICAgfSxcbiAgfVxufVxuIiwKICAgICAgICAiXG4gICAgKHdpbmRvdy5fX05FWFRfUCA9IHdpbmRvdy5fX05FWFRfUCB8fCBbXSkucHVzaChbXG4gICAgICBcIi9zdGF0aWNcIixcbiAgICAgIGZ1bmN0aW9uICgpIHtcbiAgICAgICAgcmV0dXJuIHJlcXVpcmUoXCJwcml2YXRlLW5leHQtcGFnZXMvc3RhdGljLmpzXCIpO1xuICAgICAgfVxuICAgIF0pO1xuICAgIGlmKG1vZHVsZS5ob3QpIHtcbiAgICAgIG1vZHVsZS5ob3QuZGlzcG9zZShmdW5jdGlvbiAoKSB7XG4gICAgICAgIHdpbmRvdy5fX05FWFRfUC5wdXNoKFtcIi9zdGF0aWNcIl0pXG4gICAgICB9KTtcbiAgICB9XG4gICIKICAgIF0sCiAgICAibmFtZXMiOiBbCiAgICAgICAgIlN0YXRpY1BhZ2UiLAogICAgICAgICJkYXRhIiwKICAgICAgICAiZm9vIgogICAgXQp9Cg==)
- [output (old)](https://evanw.github.io/source-map-visualization/#NjY4MgAoc2VsZi53ZWJwYWNrQ2h1bmtfTl9FID0gc2VsZi53ZWJwYWNrQ2h1bmtfTl9FIHx8IFtdKS5wdXNoKFsKICAgIFsKICAgICAgICAxNTgKICAgIF0sCiAgICB7CiAgICAgICAgMjk0MzogZnVuY3Rpb24obiwgdCwgdSkgewogICAgICAgICAgICAidXNlIHN0cmljdCI7CiAgICAgICAgICAgIHZhciByID0gZnVuY3Rpb24obikgewogICAgICAgICAgICAgICAgdmFyIHQgPSBuLmRhdGE7CiAgICAgICAgICAgICAgICByZXR1cm4gKDAsIF8uanN4KSgiZGl2IiwgewogICAgICAgICAgICAgICAgICAgIGNoaWxkcmVuOiB0LmZvbwogICAgICAgICAgICAgICAgfSk7CiAgICAgICAgICAgIH07CiAgICAgICAgICAgIHUucih0KSwgdS5kKHQsIHsKICAgICAgICAgICAgICAgIF9fTl9TU0c6IGZ1bmN0aW9uKCkgewogICAgICAgICAgICAgICAgICAgIHJldHVybiBpOwogICAgICAgICAgICAgICAgfSwKICAgICAgICAgICAgICAgIGRlZmF1bHQ6IGZ1bmN0aW9uKCkgewogICAgICAgICAgICAgICAgICAgIHJldHVybiByOwogICAgICAgICAgICAgICAgfQogICAgICAgICAgICB9KTsKICAgICAgICAgICAgdmFyIF8gPSB1KDQ1MTIpLCBpID0gITA7CiAgICAgICAgfSwKICAgICAgICA3MTM5OiBmdW5jdGlvbihuLCB0LCB1KSB7CiAgICAgICAgICAgICh3aW5kb3cuX19ORVhUX1AgPSB3aW5kb3cuX19ORVhUX1AgfHwgW10pLnB1c2goWwogICAgICAgICAgICAgICAgIi9zdGF0aWMiLAogICAgICAgICAgICAgICAgZnVuY3Rpb24oKSB7CiAgICAgICAgICAgICAgICAgICAgcmV0dXJuIHUoMjk0Myk7CiAgICAgICAgICAgICAgICB9CiAgICAgICAgICAgIF0pOwogICAgICAgIH0KICAgIH0sCiAgICBmdW5jdGlvbihuKSB7CiAgICAgICAgbi5PKDAsIFsKICAgICAgICAgICAgNzc0LAogICAgICAgICAgICA4ODgsCiAgICAgICAgICAgIDE3OQogICAgICAgIF0sIGZ1bmN0aW9uKCkgewogICAgICAgICAgICByZXR1cm4gbihuLnMgPSA3MTM5KTsKICAgICAgICB9KSwgX05fRSA9IG4uTygpOwogICAgfQpdKTsKCi8vIyBzb3VyY2VNYXBwaW5nVVJMPWRhdGE6YXBwbGljYXRpb24vanNvbjtiYXNlNjQsZXdvZ0lDSnRZWEJ3YVc1bmN5STZJQ0pCUlVGRFFTeERRVUZCUVN4TFFVRkxMR2RDUVVGdFFpeEhRVUZIUVN4TFFVRkxMR2RDUVVGdFFpeEpRVUZKTEVWQlFVVXNRVUZCUkN4RlFVRkhReXhKUVVGSkxFTkJRVU03U1VGRE4wUTdVVUZCUXp0TFFVRkpPMGxCUTB3N1VVRkRWU3hOUVVGWkxGTkJRMlJETEVOQlFYVkNMRVZCUTNaQ1F5eERRVUZ0UWl4RlFVTnVRa01zUTBGQmJVSXNSVUZEY2tJN1dVRkRSVHRuUWtaT1dFTXNTVUZCUVN4VFFVRkJReXhEUVVGQkxFVkJRVUU3WjBKQlFVRXNTVUZCUVVNc1NVRkJRVVFzUlVGQlFVTXNTVUZCUVR0blFrRkJRU3hQUVVGQkxFTkJRVUVzUjBORVJFTXNSVUZCUVVNc1IwRkJRU3hGUVVGQkxFOUJRVUU3YjBKQlEwRkRMRlZCUVVGSUxFVkJRVUZKTEVkQlFVRTdaMEpCUTBFN1dVRkRRVHRaUVV0WlVDeEZRVUZ2UWxFc1EwRkJReXhEUVVGRFZDeEpRVU5FUXl4RlFVRnZRbE1zUTBGQlF5eERRVUZEVml4SFFVRnhRanRuUWtGRGRrTlhMRk5CUVZNc1YwRkJXVHR2UWtGRGRFTXNUMEZCY1VKQk8yZENSRnBxUWp0blFrRkJjVUpETEZOQlFVRXNWMEZCUVR0dlFrRkRNME1zVDBGQlQxWTdaMEpCUVUwN1dVRkRaRHRaUVVGQkxFbEJRVUZITEVsQlFVRktMRVZCUVVFc1QwRkJRVlVzU1VGQlFTeERRVUZCTzFGRFMwMDdVVUZMVUN4TlFVRkJMRk5CUVVGYUxFTkJRVUVzUlVGQlFXTXNRMEZCUVN4RlFVRkJXaXhEUVVGQkxFVkJRVUU3V1VGQlFXRXNRMEZCUVVFc1QwRkJRVU1zVVVGQlFTeEhRVUZCUkN4UFFVRkJReXhSUVVGQkxFbEJRVUVzUlVGQlFTeEZRVUZCYWtJc1NVRkJRU3hEUVVGQk8yZENRVUZCTzJkQ1FVRkJMRmRCUVVFN2IwSkJRVUVzVDBGQlFVY3NSVUZCUVR0blFrRkJRVHRoUVVGQk8xRkJRVUU3U1VGQlFUdEpRVUZCTEZOQlFVRkJMRU5CUVVFc1JVRkJRVHRSUVVGQlFTeEZRVUZCWlN4RFFVRkJMRU5CUVVFc1IwRkJRVHRaUVVGQk8xbEJRVUU3V1VGQlFUdFRRVUZCTEVWQlFVRXNWMEZCUVR0WlFVRkJMRTlCUVVGbUxFVkJRVUZCTEVWQlFVRm5RaXhEUVVGQkxFZEJRVUU3VVVGQlFTeEpRVUZCUXl4UFFVRkJha0lzUlVGQlFXVXNRMEZCUVR0SlFVRkJPME5CUVVFaUxBb2dJQ0p1WVcxbGN5STZJRnNLSUNBZ0lDSnpaV3htSWl3S0lDQWdJQ0p3ZFhOb0lpd0tJQ0FnSUNKZlgzVnVkWE5sWkY5M1pXSndZV05yWDIxdlpIVnNaU0lzQ2lBZ0lDQWlYMTkzWldKd1lXTnJYMlY0Y0c5eWRITmZYeUlzQ2lBZ0lDQWlYMTkzWldKd1lXTnJYM0psY1hWcGNtVmZYeUlzQ2lBZ0lDQWlVM1JoZEdsalVHRm5aU0lzQ2lBZ0lDQWlYM0psWmlJc0NpQWdJQ0FpWkdGMFlTSXNDaUFnSUNBaWNtVmhZM1JmYW5ONFgzSjFiblJwYldWZlgxZEZRbEJCUTB0ZlNVMVFUMUpVUlVSZlRVOUVWVXhGWHpCZlh5SXNDaUFnSUNBaWFuTjRJaXdLSUNBZ0lDSmphR2xzWkhKbGJpSXNDaUFnSUNBaVptOXZJaXdLSUNBZ0lDSnlJaXdLSUNBZ0lDSmtJaXdLSUNBZ0lDSmZYMDVmVTFOSElpd0tJQ0FnSUNKa1pXWmhkV3gwSWl3S0lDQWdJQ0pmWDNWdWRYTmxaRjkzWldKd1lXTnJYMlY0Y0c5eWRITWlMQW9nSUNBZ0luZHBibVJ2ZHlJc0NpQWdJQ0FpWDE5T1JWaFVYMUFpTEFvZ0lDQWdJazhpTEFvZ0lDQWdJbk1pTEFvZ0lDQWdJbDlPWDBVaUNpQWdYU3dLSUNBaWMyOTFjbU5sY3lJNklGc0tJQ0FnSUNKM1pXSndZV05yT2k4dkxpOXdZV2RsY3k5emRHRjBhV011YW5NaUxBb2dJQ0FnSW5kbFluQmhZMnM2THk4dUxpOHVMaTh1TGk5d1lXTnJZV2RsY3k5dVpYaDBMMlJwYzNRdlluVnBiR1F2ZDJWaWNHRmpheTlzYjJGa1pYSnpMMjVsZUhRdFkyeHBaVzUwTFhCaFoyVnpMV3h2WVdSbGNpNXFjejl3WVdkbFBTVXlSbk4wWVhScFl5WmhZbk52YkhWMFpWQmhaMlZRWVhSb1BYQnlhWFpoZEdVdGJtVjRkQzF3WVdkbGN5VXlSbk4wWVhScFl5NXFjeUVpTEFvZ0lDQWdJaTR1THk0dUwybHVjSFYwTDJsdVpHVjRMbXB6SWdvZ0lGMHNDaUFnSW5OdmRYSmpaWE5EYjI1MFpXNTBJam9nV3dvZ0lDQWdJbVY0Y0c5eWRDQmtaV1poZFd4MElHWjFibU4wYVc5dUlGTjBZWFJwWTFCaFoyVW9leUJrWVhSaElIMHBJSHRjYmlBZ2NtVjBkWEp1SUR4a2FYWStlMlJoZEdFdVptOXZmVHd2WkdsMlBseHVmVnh1WEc1bGVIQnZjblFnWVhONWJtTWdablZ1WTNScGIyNGdaMlYwVTNSaGRHbGpVSEp2Y0hNb0tTQjdYRzRnSUhKbGRIVnliaUI3WEc0Z0lDQWdjSEp2Y0hNNklIdGNiaUFnSUNBZ0lHUmhkR0U2SUh0Y2JpQWdJQ0FnSUNBZ1ptOXZPaUFuWW1GeUp5eGNiaUFnSUNBZ0lIMHNYRzRnSUNBZ2ZTeGNiaUFnZlZ4dWZWeHVJaXdLSUNBZ0lDSmNiaUFnSUNBb2QybHVaRzkzTGw5ZlRrVllWRjlRSUQwZ2QybHVaRzkzTGw5ZlRrVllWRjlRSUh4OElGdGRLUzV3ZFhOb0tGdGNiaUFnSUNBZ0lGd2lMM04wWVhScFkxd2lMRnh1SUNBZ0lDQWdablZ1WTNScGIyNGdLQ2tnZTF4dUlDQWdJQ0FnSUNCeVpYUjFjbTRnY21WeGRXbHlaU2hjSW5CeWFYWmhkR1V0Ym1WNGRDMXdZV2RsY3k5emRHRjBhV011YW5OY0lpazdYRzRnSUNBZ0lDQjlYRzRnSUNBZ1hTazdYRzRnSUNBZ2FXWW9iVzlrZFd4bExtaHZkQ2tnZTF4dUlDQWdJQ0FnYlc5a2RXeGxMbWh2ZEM1a2FYTndiM05sS0daMWJtTjBhVzl1SUNncElIdGNiaUFnSUNBZ0lDQWdkMmx1Wkc5M0xsOWZUa1ZZVkY5UUxuQjFjMmdvVzF3aUwzTjBZWFJwWTF3aVhTbGNiaUFnSUNBZ0lIMHBPMXh1SUNBZ0lIMWNiaUFnSWl3S0lDQWdJQ0lvYzJWc1psdGNJbmRsWW5CaFkydERhSFZ1YTE5T1gwVmNJbDBnUFNCelpXeG1XMXdpZDJWaWNHRmphME5vZFc1clgwNWZSVndpWFNCOGZDQmJYU2t1Y0hWemFDaGJYRzRnSUNBZ1d6RTFPRjBzWEc0Z0lDQWdlMXh1SUNBZ0lDQWdJQ0F2S2lvcUx5QXlPVFF6T2lBdktpb3FMeUJtZFc1amRHbHZiaUFvWEc0Z0lDQWdJQ0FnSUNBZ0lDQmZYM1Z1ZFhObFpGOTNaV0p3WVdOclgyMXZaSFZzWlN4Y2JpQWdJQ0FnSUNBZ0lDQWdJRjlmZDJWaWNHRmphMTlsZUhCdmNuUnpYMThzWEc0Z0lDQWdJQ0FnSUNBZ0lDQmZYM2RsWW5CaFkydGZjbVZ4ZFdseVpWOWZYRzRnSUNBZ0lDQWdJQ2tnZTF4dUlDQWdJQ0FnSUNBZ0lDQWdYQ0oxYzJVZ2MzUnlhV04wWENJN1hHNGdJQ0FnSUNBZ0lDQWdJQ0JmWDNkbFluQmhZMnRmY21WeGRXbHlaVjlmTG5Jb1gxOTNaV0p3WVdOclgyVjRjRzl5ZEhOZlh5azdYRzRnSUNBZ0lDQWdJQ0FnSUNBdktpQm9ZWEp0YjI1NUlHVjRjRzl5ZENBcUx5QmZYM2RsWW5CaFkydGZjbVZ4ZFdseVpWOWZMbVFvWDE5M1pXSndZV05yWDJWNGNHOXlkSE5mWHl3Z2UxeHVJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ0lDOHFJR2hoY20xdmJua2daWGh3YjNKMElDb3ZJRjlmVGw5VFUwYzZJR1oxYm1OMGFXOXVJQ2dwSUh0Y2JpQWdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdjbVYwZFhKdUlDOHFJR0pwYm1ScGJtY2dLaThnWDE5T1gxTlRSenRjYmlBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0I5TEZ4dUlDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUM4cUlHaGhjbTF2Ym5rZ1pYaHdiM0owSUNvdklHUmxabUYxYkhRNklHWjFibU4wYVc5dUlDZ3BJSHRjYmlBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ2NtVjBkWEp1SUM4cUlHSnBibVJwYm1jZ0tpOGdVM1JoZEdsalVHRm5aVHRjYmlBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0I5TEZ4dUlDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUM4cUlHaGhjbTF2Ym5rZ1pYaHdiM0owSUNvdlhHNGdJQ0FnSUNBZ0lDQWdJQ0I5S1R0Y2JpQWdJQ0FnSUNBZ0lDQWdJQzhxSUdoaGNtMXZibmtnYVcxd2IzSjBJQ292SUhaaGNpQnlaV0ZqZEY5cWMzaGZjblZ1ZEdsdFpWOWZWMFZDVUVGRFMxOUpUVkJQVWxSRlJGOU5UMFJWVEVWZk1GOWZJRDFjYmlBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0JmWDNkbFluQmhZMnRmY21WeGRXbHlaVjlmS0RRMU1USXBPMXh1WEc0Z0lDQWdJQ0FnSUNBZ0lDQjJZWElnWDE5T1gxTlRSeUE5SUhSeWRXVTdYRzRnSUNBZ0lDQWdJQ0FnSUNCbWRXNWpkR2x2YmlCVGRHRjBhV05RWVdkbEtGOXlaV1lwSUh0Y2JpQWdJQ0FnSUNBZ0lDQWdJQ0FnSUNCMllYSWdaR0YwWVNBOUlGOXlaV1l1WkdGMFlUdGNiaUFnSUNBZ0lDQWdJQ0FnSUNBZ0lDQnlaWFIxY200Z0x5b2pYMTlRVlZKRlgxOHFMeUFvTUN4Y2JpQWdJQ0FnSUNBZ0lDQWdJQ0FnSUNCeVpXRmpkRjlxYzNoZmNuVnVkR2x0WlY5ZlYwVkNVRUZEUzE5SlRWQlBVbFJGUkY5TlQwUlZURVZmTUY5ZkxtcHplQ2tvWENKa2FYWmNJaXdnZTF4dUlDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ0lDQmphR2xzWkhKbGJqb2daR0YwWVM1bWIyOHNYRzRnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdmU2s3WEc0Z0lDQWdJQ0FnSUNBZ0lDQjlYRzVjYmlBZ0lDQWdJQ0FnSUNBZ0lDOHFLaW92WEc0Z0lDQWdJQ0FnSUgwc1hHNWNiaUFnSUNBZ0lDQWdMeW9xS2k4Z056RXpPVG9nTHlvcUtpOGdablZ1WTNScGIyNGdLRnh1SUNBZ0lDQWdJQ0FnSUNBZ1gxOTFiblZ6WldSZmQyVmljR0ZqYTE5dGIyUjFiR1VzWEc0Z0lDQWdJQ0FnSUNBZ0lDQmZYM1Z1ZFhObFpGOTNaV0p3WVdOclgyVjRjRzl5ZEhNc1hHNGdJQ0FnSUNBZ0lDQWdJQ0JmWDNkbFluQmhZMnRmY21WeGRXbHlaVjlmWEc0Z0lDQWdJQ0FnSUNrZ2UxeHVJQ0FnSUNBZ0lDQWdJQ0FnS0hkcGJtUnZkeTVmWDA1RldGUmZVQ0E5SUhkcGJtUnZkeTVmWDA1RldGUmZVQ0I4ZkNCYlhTa3VjSFZ6YUNoYlhHNGdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ1hDSXZjM1JoZEdsalhDSXNYRzRnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdablZ1WTNScGIyNGdLQ2tnZTF4dUlDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ0lDQnlaWFIxY200Z1gxOTNaV0p3WVdOclgzSmxjWFZwY21WZlh5Z3lPVFF6S1R0Y2JpQWdJQ0FnSUNBZ0lDQWdJQ0FnSUNCOUxGeHVJQ0FnSUNBZ0lDQWdJQ0FnWFNrN1hHNGdJQ0FnSUNBZ0lDQWdJQ0JwWmlBb1ptRnNjMlVwSUh0Y2JpQWdJQ0FnSUNBZ0lDQWdJSDFjYmx4dUlDQWdJQ0FnSUNBZ0lDQWdMeW9xS2k5Y2JpQWdJQ0FnSUNBZ2ZTeGNiaUFnSUNCOUxGeHVJQ0FnSUM4cUtpb3FLaW92SUdaMWJtTjBhVzl1SUNoZlgzZGxZbkJoWTJ0ZmNtVnhkV2x5WlY5ZktTQjdYRzRnSUNBZ0lDQWdJQzh2SUhkbFluQmhZMnRTZFc1MGFXMWxUVzlrZFd4bGMxeHVJQ0FnSUNBZ0lDQXZLaW9xS2lvcUx5QjJZWElnWDE5M1pXSndZV05yWDJWNFpXTmZYeUE5SUdaMWJtTjBhVzl1SUNodGIyUjFiR1ZKWkNrZ2UxeHVJQ0FnSUNBZ0lDQWdJQ0FnY21WMGRYSnVJRjlmZDJWaWNHRmphMTl5WlhGMWFYSmxYMThvS0Y5ZmQyVmljR0ZqYTE5eVpYRjFhWEpsWDE4dWN5QTlJRzF2WkhWc1pVbGtLU2s3WEc0Z0lDQWdJQ0FnSUgwN1hHNGdJQ0FnSUNBZ0lDOHFLaW9xS2lvdklGOWZkMlZpY0dGamExOXlaWEYxYVhKbFgxOHVUeWd3TENCYk56YzBMQ0E0T0Rnc0lERTNPVjBzSUdaMWJtTjBhVzl1SUNncElIdGNiaUFnSUNBZ0lDQWdJQ0FnSUhKbGRIVnliaUJmWDNkbFluQmhZMnRmWlhobFkxOWZLRGN4TXprcE8xeHVJQ0FnSUNBZ0lDQjlLVHRjYmlBZ0lDQWdJQ0FnTHlvcUtpb3FLaThnZG1GeUlGOWZkMlZpY0dGamExOWxlSEJ2Y25SelgxOGdQU0JmWDNkbFluQmhZMnRmY21WeGRXbHlaVjlmTGs4b0tUdGNiaUFnSUNBZ0lDQWdMeW9xS2lvcUtpOGdYMDVmUlNBOUlGOWZkMlZpY0dGamExOWxlSEJ2Y25SelgxODdYRzRnSUNBZ0lDQWdJQzhxS2lvcUtpb3ZYRzRnSUNBZ2ZTeGNibDBwTzF4dUlnb2dJRjBzQ2lBZ0luWmxjbk5wYjI0aU9pQXpDbjA9NDIxMQB7CiAgIm1hcHBpbmdzIjogIkFFQUNBLENBQUFBLEtBQUssZ0JBQW1CLEdBQUdBLEtBQUssZ0JBQW1CLElBQUksRUFBRSxBQUFELEVBQUdDLElBQUksQ0FBQztJQUM3RDtRQUFDO0tBQUk7SUFDTDtRQUNVLE1BQVksU0FDZEMsQ0FBdUIsRUFDdkJDLENBQW1CLEVBQ25CQyxDQUFtQixFQUNyQjtZQUNFO2dCRk5YQyxJQUFBLFNBQUFDLENBQUEsRUFBQTtnQkFBQSxJQUFBQyxJQUFBRCxFQUFBQyxJQUFBO2dCQUFBLE9BQUEsQ0FBQSxHQ0REQyxFQUFBQyxHQUFBLEVBQUEsT0FBQTtvQkFDQUMsVUFBQUgsRUFBQUksR0FBQTtnQkFDQTtZQUNBO1lBS1lQLEVBQW9CUSxDQUFDLENBQUNULElBQ0RDLEVBQW9CUyxDQUFDLENBQUNWLEdBQXFCO2dCQUN2Q1csU0FBUyxXQUFZO29CQUN0QyxPQUFxQkE7Z0JEWmpCO2dCQUFxQkMsU0FBQSxXQUFBO29CQUMzQyxPQUFPVjtnQkFBTTtZQUNkO1lBQUEsSUFBQUcsSUFBQUosRUFBQSxPQUFBVSxJQUFBLENBQUE7UUNLTTtRQUtQLE1BQUEsU0FBQVosQ0FBQSxFQUFBYyxDQUFBLEVBQUFaLENBQUEsRUFBQTtZQUFBYSxDQUFBQSxPQUFBQyxRQUFBLEdBQUFELE9BQUFDLFFBQUEsSUFBQSxFQUFBLEVBQUFqQixJQUFBLENBQUE7Z0JBQUE7Z0JBQUEsV0FBQTtvQkFBQSxPQUFBRyxFQUFBO2dCQUFBO2FBQUE7UUFBQTtJQUFBO0lBQUEsU0FBQUEsQ0FBQSxFQUFBO1FBQUFBLEVBQUFlLENBQUEsQ0FBQSxHQUFBO1lBQUE7WUFBQTtZQUFBO1NBQUEsRUFBQSxXQUFBO1lBQUEsT0FBQWYsRUFBQUEsRUFBQWdCLENBQUEsR0FBQTtRQUFBLElBQUFDLE9BQUFqQixFQUFBZSxDQUFBO0lBQUE7Q0FBQSIsCiAgIm5hbWVzIjogWwogICAgInNlbGYiLAogICAgInB1c2giLAogICAgIl9fdW51c2VkX3dlYnBhY2tfbW9kdWxlIiwKICAgICJfX3dlYnBhY2tfZXhwb3J0c19fIiwKICAgICJfX3dlYnBhY2tfcmVxdWlyZV9fIiwKICAgICJTdGF0aWNQYWdlIiwKICAgICJfcmVmIiwKICAgICJkYXRhIiwKICAgICJyZWFjdF9qc3hfcnVudGltZV9fV0VCUEFDS19JTVBPUlRFRF9NT0RVTEVfMF9fIiwKICAgICJqc3giLAogICAgImNoaWxkcmVuIiwKICAgICJmb28iLAogICAgInIiLAogICAgImQiLAogICAgIl9fTl9TU0ciLAogICAgImRlZmF1bHQiLAogICAgIl9fdW51c2VkX3dlYnBhY2tfZXhwb3J0cyIsCiAgICAid2luZG93IiwKICAgICJfX05FWFRfUCIsCiAgICAiTyIsCiAgICAicyIsCiAgICAiX05fRSIKICBdLAogICJzb3VyY2VzIjogWwogICAgIndlYnBhY2s6Ly8uL3BhZ2VzL3N0YXRpYy5qcyIsCiAgICAid2VicGFjazovLy4uLy4uLy4uL3BhY2thZ2VzL25leHQvZGlzdC9idWlsZC93ZWJwYWNrL2xvYWRlcnMvbmV4dC1jbGllbnQtcGFnZXMtbG9hZGVyLmpzP3BhZ2U9JTJGc3RhdGljJmFic29sdXRlUGFnZVBhdGg9cHJpdmF0ZS1uZXh0LXBhZ2VzJTJGc3RhdGljLmpzISIsCiAgICAiLi4vLi4vaW5wdXQvaW5kZXguanMiCiAgXSwKICAic291cmNlc0NvbnRlbnQiOiBbCiAgICAiZXhwb3J0IGRlZmF1bHQgZnVuY3Rpb24gU3RhdGljUGFnZSh7IGRhdGEgfSkge1xuICByZXR1cm4gPGRpdj57ZGF0YS5mb299PC9kaXY+XG59XG5cbmV4cG9ydCBhc3luYyBmdW5jdGlvbiBnZXRTdGF0aWNQcm9wcygpIHtcbiAgcmV0dXJuIHtcbiAgICBwcm9wczoge1xuICAgICAgZGF0YToge1xuICAgICAgICBmb286ICdiYXInLFxuICAgICAgfSxcbiAgICB9LFxuICB9XG59XG4iLAogICAgIlxuICAgICh3aW5kb3cuX19ORVhUX1AgPSB3aW5kb3cuX19ORVhUX1AgfHwgW10pLnB1c2goW1xuICAgICAgXCIvc3RhdGljXCIsXG4gICAgICBmdW5jdGlvbiAoKSB7XG4gICAgICAgIHJldHVybiByZXF1aXJlKFwicHJpdmF0ZS1uZXh0LXBhZ2VzL3N0YXRpYy5qc1wiKTtcbiAgICAgIH1cbiAgICBdKTtcbiAgICBpZihtb2R1bGUuaG90KSB7XG4gICAgICBtb2R1bGUuaG90LmRpc3Bvc2UoZnVuY3Rpb24gKCkge1xuICAgICAgICB3aW5kb3cuX19ORVhUX1AucHVzaChbXCIvc3RhdGljXCJdKVxuICAgICAgfSk7XG4gICAgfVxuICAiLAogICAgIihzZWxmW1wid2VicGFja0NodW5rX05fRVwiXSA9IHNlbGZbXCJ3ZWJwYWNrQ2h1bmtfTl9FXCJdIHx8IFtdKS5wdXNoKFtcbiAgICBbMTU4XSxcbiAgICB7XG4gICAgICAgIC8qKiovIDI5NDM6IC8qKiovIGZ1bmN0aW9uIChcbiAgICAgICAgICAgIF9fdW51c2VkX3dlYnBhY2tfbW9kdWxlLFxuICAgICAgICAgICAgX193ZWJwYWNrX2V4cG9ydHNfXyxcbiAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX19cbiAgICAgICAgKSB7XG4gICAgICAgICAgICBcInVzZSBzdHJpY3RcIjtcbiAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX18ucihfX3dlYnBhY2tfZXhwb3J0c19fKTtcbiAgICAgICAgICAgIC8qIGhhcm1vbnkgZXhwb3J0ICovIF9fd2VicGFja19yZXF1aXJlX18uZChfX3dlYnBhY2tfZXhwb3J0c19fLCB7XG4gICAgICAgICAgICAgICAgLyogaGFybW9ueSBleHBvcnQgKi8gX19OX1NTRzogZnVuY3Rpb24gKCkge1xuICAgICAgICAgICAgICAgICAgICByZXR1cm4gLyogYmluZGluZyAqLyBfX05fU1NHO1xuICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICAgICAgLyogaGFybW9ueSBleHBvcnQgKi8gZGVmYXVsdDogZnVuY3Rpb24gKCkge1xuICAgICAgICAgICAgICAgICAgICByZXR1cm4gLyogYmluZGluZyAqLyBTdGF0aWNQYWdlO1xuICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICAgICAgLyogaGFybW9ueSBleHBvcnQgKi9cbiAgICAgICAgICAgIH0pO1xuICAgICAgICAgICAgLyogaGFybW9ueSBpbXBvcnQgKi8gdmFyIHJlYWN0X2pzeF9ydW50aW1lX19XRUJQQUNLX0lNUE9SVEVEX01PRFVMRV8wX18gPVxuICAgICAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX18oNDUxMik7XG5cbiAgICAgICAgICAgIHZhciBfX05fU1NHID0gdHJ1ZTtcbiAgICAgICAgICAgIGZ1bmN0aW9uIFN0YXRpY1BhZ2UoX3JlZikge1xuICAgICAgICAgICAgICAgIHZhciBkYXRhID0gX3JlZi5kYXRhO1xuICAgICAgICAgICAgICAgIHJldHVybiAvKiNfX1BVUkVfXyovICgwLFxuICAgICAgICAgICAgICAgIHJlYWN0X2pzeF9ydW50aW1lX19XRUJQQUNLX0lNUE9SVEVEX01PRFVMRV8wX18uanN4KShcImRpdlwiLCB7XG4gICAgICAgICAgICAgICAgICAgIGNoaWxkcmVuOiBkYXRhLmZvbyxcbiAgICAgICAgICAgICAgICB9KTtcbiAgICAgICAgICAgIH1cblxuICAgICAgICAgICAgLyoqKi9cbiAgICAgICAgfSxcblxuICAgICAgICAvKioqLyA3MTM5OiAvKioqLyBmdW5jdGlvbiAoXG4gICAgICAgICAgICBfX3VudXNlZF93ZWJwYWNrX21vZHVsZSxcbiAgICAgICAgICAgIF9fdW51c2VkX3dlYnBhY2tfZXhwb3J0cyxcbiAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX19cbiAgICAgICAgKSB7XG4gICAgICAgICAgICAod2luZG93Ll9fTkVYVF9QID0gd2luZG93Ll9fTkVYVF9QIHx8IFtdKS5wdXNoKFtcbiAgICAgICAgICAgICAgICBcIi9zdGF0aWNcIixcbiAgICAgICAgICAgICAgICBmdW5jdGlvbiAoKSB7XG4gICAgICAgICAgICAgICAgICAgIHJldHVybiBfX3dlYnBhY2tfcmVxdWlyZV9fKDI5NDMpO1xuICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICBdKTtcbiAgICAgICAgICAgIGlmIChmYWxzZSkge1xuICAgICAgICAgICAgfVxuXG4gICAgICAgICAgICAvKioqL1xuICAgICAgICB9LFxuICAgIH0sXG4gICAgLyoqKioqKi8gZnVuY3Rpb24gKF9fd2VicGFja19yZXF1aXJlX18pIHtcbiAgICAgICAgLy8gd2VicGFja1J1bnRpbWVNb2R1bGVzXG4gICAgICAgIC8qKioqKiovIHZhciBfX3dlYnBhY2tfZXhlY19fID0gZnVuY3Rpb24gKG1vZHVsZUlkKSB7XG4gICAgICAgICAgICByZXR1cm4gX193ZWJwYWNrX3JlcXVpcmVfXygoX193ZWJwYWNrX3JlcXVpcmVfXy5zID0gbW9kdWxlSWQpKTtcbiAgICAgICAgfTtcbiAgICAgICAgLyoqKioqKi8gX193ZWJwYWNrX3JlcXVpcmVfXy5PKDAsIFs3NzQsIDg4OCwgMTc5XSwgZnVuY3Rpb24gKCkge1xuICAgICAgICAgICAgcmV0dXJuIF9fd2VicGFja19leGVjX18oNzEzOSk7XG4gICAgICAgIH0pO1xuICAgICAgICAvKioqKioqLyB2YXIgX193ZWJwYWNrX2V4cG9ydHNfXyA9IF9fd2VicGFja19yZXF1aXJlX18uTygpO1xuICAgICAgICAvKioqKioqLyBfTl9FID0gX193ZWJwYWNrX2V4cG9ydHNfXztcbiAgICAgICAgLyoqKioqKi9cbiAgICB9LFxuXSk7XG4iCiAgXSwKICAidmVyc2lvbiI6IDMKfQ==)
- [output (new)](https://evanw.github.io/source-map-visualization/#NTY0NgAoc2VsZi53ZWJwYWNrQ2h1bmtfTl9FID0gc2VsZi53ZWJwYWNrQ2h1bmtfTl9FIHx8IFtdKS5wdXNoKFsKICAgIFsKICAgICAgICAxNTgKICAgIF0sCiAgICB7CiAgICAgICAgMjk0MzogZnVuY3Rpb24obiwgdCwgdSkgewogICAgICAgICAgICAidXNlIHN0cmljdCI7CiAgICAgICAgICAgIHZhciByID0gZnVuY3Rpb24obikgewogICAgICAgICAgICAgICAgdmFyIHQgPSBuLmRhdGE7CiAgICAgICAgICAgICAgICByZXR1cm4gKDAsIF8uanN4KSgiZGl2IiwgewogICAgICAgICAgICAgICAgICAgIGNoaWxkcmVuOiB0LmZvbwogICAgICAgICAgICAgICAgfSk7CiAgICAgICAgICAgIH07CiAgICAgICAgICAgIHUucih0KSwgdS5kKHQsIHsKICAgICAgICAgICAgICAgIF9fTl9TU0c6IGZ1bmN0aW9uKCkgewogICAgICAgICAgICAgICAgICAgIHJldHVybiBpOwogICAgICAgICAgICAgICAgfSwKICAgICAgICAgICAgICAgIGRlZmF1bHQ6IGZ1bmN0aW9uKCkgewogICAgICAgICAgICAgICAgICAgIHJldHVybiByOwogICAgICAgICAgICAgICAgfQogICAgICAgICAgICB9KTsKICAgICAgICAgICAgdmFyIF8gPSB1KDQ1MTIpLCBpID0gITA7CiAgICAgICAgfSwKICAgICAgICA3MTM5OiBmdW5jdGlvbihuLCB0LCB1KSB7CiAgICAgICAgICAgICh3aW5kb3cuX19ORVhUX1AgPSB3aW5kb3cuX19ORVhUX1AgfHwgW10pLnB1c2goWwogICAgICAgICAgICAgICAgIi9zdGF0aWMiLAogICAgICAgICAgICAgICAgZnVuY3Rpb24oKSB7CiAgICAgICAgICAgICAgICAgICAgcmV0dXJuIHUoMjk0Myk7CiAgICAgICAgICAgICAgICB9CiAgICAgICAgICAgIF0pOwogICAgICAgIH0KICAgIH0sCiAgICBmdW5jdGlvbihuKSB7CiAgICAgICAgbi5PKDAsIFsKICAgICAgICAgICAgNzc0LAogICAgICAgICAgICA4ODgsCiAgICAgICAgICAgIDE3OQogICAgICAgIF0sIGZ1bmN0aW9uKCkgewogICAgICAgICAgICByZXR1cm4gbihuLnMgPSA3MTM5KTsKICAgICAgICB9KSwgX05fRSA9IG4uTygpOwogICAgfQpdKTsKCi8vIyBzb3VyY2VNYXBwaW5nVVJMPWRhdGE6YXBwbGljYXRpb24vanNvbjtiYXNlNjQsZXdvZ0lDSnRZWEJ3YVc1bmN5STZJQ0k3T3p0TFFVTlRJaXdLSUNBaWJtRnRaWE1pT2lCYkNpQWdJQ0FpYzJWc1ppSXNDaUFnSUNBaWNIVnphQ0lzQ2lBZ0lDQWlYMTkxYm5WelpXUmZkMlZpY0dGamExOXRiMlIxYkdVaUxBb2dJQ0FnSWw5ZmQyVmljR0ZqYTE5bGVIQnZjblJ6WDE4aUxBb2dJQ0FnSWw5ZmQyVmljR0ZqYTE5eVpYRjFhWEpsWDE4aUxBb2dJQ0FnSWxOMFlYUnBZMUJoWjJVaUxBb2dJQ0FnSWw5eVpXWWlMQW9nSUNBZ0ltUmhkR0VpTEFvZ0lDQWdJbkpsWVdOMFgycHplRjl5ZFc1MGFXMWxYMTlYUlVKUVFVTkxYMGxOVUU5U1ZFVkVYMDFQUkZWTVJWOHdYMThpTEFvZ0lDQWdJbXB6ZUNJc0NpQWdJQ0FpWTJocGJHUnlaVzRpTEFvZ0lDQWdJbVp2YnlJc0NpQWdJQ0FpY2lJc0NpQWdJQ0FpWkNJc0NpQWdJQ0FpWDE5T1gxTlRSeUlzQ2lBZ0lDQWlaR1ZtWVhWc2RDSXNDaUFnSUNBaVgxOTFiblZ6WldSZmQyVmljR0ZqYTE5bGVIQnZjblJ6SWl3S0lDQWdJQ0ozYVc1a2IzY2lMQW9nSUNBZ0lsOWZUa1ZZVkY5UUlpd0tJQ0FnSUNKUElpd0tJQ0FnSUNKeklpd0tJQ0FnSUNKZlRsOUZJZ29nSUYwc0NpQWdJbk52ZFhKalpYTWlPaUJiQ2lBZ0lDQWlkMlZpY0dGamF6b3ZMeTR2Y0dGblpYTXZjM1JoZEdsakxtcHpJaXdLSUNBZ0lDSjNaV0p3WVdOck9pOHZMaTR2TGk0dkxpNHZjR0ZqYTJGblpYTXZibVY0ZEM5a2FYTjBMMkoxYVd4a0wzZGxZbkJoWTJzdmJHOWhaR1Z5Y3k5dVpYaDBMV05zYVdWdWRDMXdZV2RsY3kxc2IyRmtaWEl1YW5NL2NHRm5aVDBsTWtaemRHRjBhV01tWVdKemIyeDFkR1ZRWVdkbFVHRjBhRDF3Y21sMllYUmxMVzVsZUhRdGNHRm5aWE1sTWtaemRHRjBhV011YW5NaElpd0tJQ0FnSUNJdUxpOHVMaTlwYm5CMWRDOXBibVJsZUM1cWN5SUtJQ0JkTEFvZ0lDSnpiM1Z5WTJWelEyOXVkR1Z1ZENJNklGc0tJQ0FnSUNKbGVIQnZjblFnWkdWbVlYVnNkQ0JtZFc1amRHbHZiaUJUZEdGMGFXTlFZV2RsS0hzZ1pHRjBZU0I5S1NCN1hHNGdJSEpsZEhWeWJpQThaR2wyUG50a1lYUmhMbVp2YjMwOEwyUnBkajVjYm4xY2JseHVaWGh3YjNKMElHRnplVzVqSUdaMWJtTjBhVzl1SUdkbGRGTjBZWFJwWTFCeWIzQnpLQ2tnZTF4dUlDQnlaWFIxY200Z2UxeHVJQ0FnSUhCeWIzQnpPaUI3WEc0Z0lDQWdJQ0JrWVhSaE9pQjdYRzRnSUNBZ0lDQWdJR1p2YnpvZ0oySmhjaWNzWEc0Z0lDQWdJQ0I5TEZ4dUlDQWdJSDBzWEc0Z0lIMWNibjFjYmlJc0NpQWdJQ0FpWEc0Z0lDQWdLSGRwYm1SdmR5NWZYMDVGV0ZSZlVDQTlJSGRwYm1SdmR5NWZYMDVGV0ZSZlVDQjhmQ0JiWFNrdWNIVnphQ2hiWEc0Z0lDQWdJQ0JjSWk5emRHRjBhV05jSWl4Y2JpQWdJQ0FnSUdaMWJtTjBhVzl1SUNncElIdGNiaUFnSUNBZ0lDQWdjbVYwZFhKdUlISmxjWFZwY21Vb1hDSndjbWwyWVhSbExXNWxlSFF0Y0dGblpYTXZjM1JoZEdsakxtcHpYQ0lwTzF4dUlDQWdJQ0FnZlZ4dUlDQWdJRjBwTzF4dUlDQWdJR2xtS0cxdlpIVnNaUzVvYjNRcElIdGNiaUFnSUNBZ0lHMXZaSFZzWlM1b2IzUXVaR2x6Y0c5elpTaG1kVzVqZEdsdmJpQW9LU0I3WEc0Z0lDQWdJQ0FnSUhkcGJtUnZkeTVmWDA1RldGUmZVQzV3ZFhOb0tGdGNJaTl6ZEdGMGFXTmNJbDBwWEc0Z0lDQWdJQ0I5S1R0Y2JpQWdJQ0I5WEc0Z0lDSXNDaUFnSUNBaUtITmxiR1piWENKM1pXSndZV05yUTJoMWJtdGZUbDlGWENKZElEMGdjMlZzWmx0Y0luZGxZbkJoWTJ0RGFIVnVhMTlPWDBWY0lsMGdmSHdnVzEwcExuQjFjMmdvVzF4dUlDQWdJRnN4TlRoZExGeHVJQ0FnSUh0Y2JpQWdJQ0FnSUNBZ0x5b3FLaThnTWprME16b2dMeW9xS2k4Z1puVnVZM1JwYjI0Z0tGeHVJQ0FnSUNBZ0lDQWdJQ0FnWDE5MWJuVnpaV1JmZDJWaWNHRmphMTl0YjJSMWJHVXNYRzRnSUNBZ0lDQWdJQ0FnSUNCZlgzZGxZbkJoWTJ0ZlpYaHdiM0owYzE5ZkxGeHVJQ0FnSUNBZ0lDQWdJQ0FnWDE5M1pXSndZV05yWDNKbGNYVnBjbVZmWDF4dUlDQWdJQ0FnSUNBcElIdGNiaUFnSUNBZ0lDQWdJQ0FnSUZ3aWRYTmxJSE4wY21samRGd2lPMXh1SUNBZ0lDQWdJQ0FnSUNBZ1gxOTNaV0p3WVdOclgzSmxjWFZwY21WZlh5NXlLRjlmZDJWaWNHRmphMTlsZUhCdmNuUnpYMThwTzF4dUlDQWdJQ0FnSUNBZ0lDQWdMeW9nYUdGeWJXOXVlU0JsZUhCdmNuUWdLaThnWDE5M1pXSndZV05yWDNKbGNYVnBjbVZmWHk1a0tGOWZkMlZpY0dGamExOWxlSEJ2Y25SelgxOHNJSHRjYmlBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0F2S2lCb1lYSnRiMjU1SUdWNGNHOXlkQ0FxTHlCZlgwNWZVMU5IT2lCbWRXNWpkR2x2YmlBb0tTQjdYRzRnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUhKbGRIVnliaUF2S2lCaWFXNWthVzVuSUNvdklGOWZUbDlUVTBjN1hHNGdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ2ZTeGNiaUFnSUNBZ0lDQWdJQ0FnSUNBZ0lDQXZLaUJvWVhKdGIyNTVJR1Y0Y0c5eWRDQXFMeUJrWldaaGRXeDBPaUJtZFc1amRHbHZiaUFvS1NCN1hHNGdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdJSEpsZEhWeWJpQXZLaUJpYVc1a2FXNW5JQ292SUZOMFlYUnBZMUJoWjJVN1hHNGdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ2ZTeGNiaUFnSUNBZ0lDQWdJQ0FnSUNBZ0lDQXZLaUJvWVhKdGIyNTVJR1Y0Y0c5eWRDQXFMMXh1SUNBZ0lDQWdJQ0FnSUNBZ2ZTazdYRzRnSUNBZ0lDQWdJQ0FnSUNBdktpQm9ZWEp0YjI1NUlHbHRjRzl5ZENBcUx5QjJZWElnY21WaFkzUmZhbk40WDNKMWJuUnBiV1ZmWDFkRlFsQkJRMHRmU1UxUVQxSlVSVVJmVFU5RVZVeEZYekJmWHlBOVhHNGdJQ0FnSUNBZ0lDQWdJQ0FnSUNBZ1gxOTNaV0p3WVdOclgzSmxjWFZwY21WZlh5ZzBOVEV5S1R0Y2JseHVJQ0FnSUNBZ0lDQWdJQ0FnZG1GeUlGOWZUbDlUVTBjZ1BTQjBjblZsTzF4dUlDQWdJQ0FnSUNBZ0lDQWdablZ1WTNScGIyNGdVM1JoZEdsalVHRm5aU2hmY21WbUtTQjdYRzRnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdkbUZ5SUdSaGRHRWdQU0JmY21WbUxtUmhkR0U3WEc0Z0lDQWdJQ0FnSUNBZ0lDQWdJQ0FnY21WMGRYSnVJQzhxSTE5ZlVGVlNSVjlmS2k4Z0tEQXNYRzRnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdjbVZoWTNSZmFuTjRYM0oxYm5ScGJXVmZYMWRGUWxCQlEwdGZTVTFRVDFKVVJVUmZUVTlFVlV4Rlh6QmZYeTVxYzNncEtGd2laR2wyWENJc0lIdGNiaUFnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0FnWTJocGJHUnlaVzQ2SUdSaGRHRXVabTl2TEZ4dUlDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUgwcE8xeHVJQ0FnSUNBZ0lDQWdJQ0FnZlZ4dVhHNGdJQ0FnSUNBZ0lDQWdJQ0F2S2lvcUwxeHVJQ0FnSUNBZ0lDQjlMRnh1WEc0Z0lDQWdJQ0FnSUM4cUtpb3ZJRGN4TXprNklDOHFLaW92SUdaMWJtTjBhVzl1SUNoY2JpQWdJQ0FnSUNBZ0lDQWdJRjlmZFc1MWMyVmtYM2RsWW5CaFkydGZiVzlrZFd4bExGeHVJQ0FnSUNBZ0lDQWdJQ0FnWDE5MWJuVnpaV1JmZDJWaWNHRmphMTlsZUhCdmNuUnpMRnh1SUNBZ0lDQWdJQ0FnSUNBZ1gxOTNaV0p3WVdOclgzSmxjWFZwY21WZlgxeHVJQ0FnSUNBZ0lDQXBJSHRjYmlBZ0lDQWdJQ0FnSUNBZ0lDaDNhVzVrYjNjdVgxOU9SVmhVWDFBZ1BTQjNhVzVrYjNjdVgxOU9SVmhVWDFBZ2ZId2dXMTBwTG5CMWMyZ29XMXh1SUNBZ0lDQWdJQ0FnSUNBZ0lDQWdJRndpTDNOMFlYUnBZMXdpTEZ4dUlDQWdJQ0FnSUNBZ0lDQWdJQ0FnSUdaMWJtTjBhVzl1SUNncElIdGNiaUFnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdJQ0FnY21WMGRYSnVJRjlmZDJWaWNHRmphMTl5WlhGMWFYSmxYMThvTWprME15azdYRzRnSUNBZ0lDQWdJQ0FnSUNBZ0lDQWdmU3hjYmlBZ0lDQWdJQ0FnSUNBZ0lGMHBPMXh1SUNBZ0lDQWdJQ0FnSUNBZ2FXWWdLR1poYkhObEtTQjdYRzRnSUNBZ0lDQWdJQ0FnSUNCOVhHNWNiaUFnSUNBZ0lDQWdJQ0FnSUM4cUtpb3ZYRzRnSUNBZ0lDQWdJSDBzWEc0Z0lDQWdmU3hjYmlBZ0lDQXZLaW9xS2lvcUx5Qm1kVzVqZEdsdmJpQW9YMTkzWldKd1lXTnJYM0psY1hWcGNtVmZYeWtnZTF4dUlDQWdJQ0FnSUNBdkx5QjNaV0p3WVdOclVuVnVkR2x0WlUxdlpIVnNaWE5jYmlBZ0lDQWdJQ0FnTHlvcUtpb3FLaThnZG1GeUlGOWZkMlZpY0dGamExOWxlR1ZqWDE4Z1BTQm1kVzVqZEdsdmJpQW9iVzlrZFd4bFNXUXBJSHRjYmlBZ0lDQWdJQ0FnSUNBZ0lISmxkSFZ5YmlCZlgzZGxZbkJoWTJ0ZmNtVnhkV2x5WlY5ZktDaGZYM2RsWW5CaFkydGZjbVZ4ZFdseVpWOWZMbk1nUFNCdGIyUjFiR1ZKWkNrcE8xeHVJQ0FnSUNBZ0lDQjlPMXh1SUNBZ0lDQWdJQ0F2S2lvcUtpb3FMeUJmWDNkbFluQmhZMnRmY21WeGRXbHlaVjlmTGs4b01Dd2dXemMzTkN3Z09EZzRMQ0F4TnpsZExDQm1kVzVqZEdsdmJpQW9LU0I3WEc0Z0lDQWdJQ0FnSUNBZ0lDQnlaWFIxY200Z1gxOTNaV0p3WVdOclgyVjRaV05mWHlnM01UTTVLVHRjYmlBZ0lDQWdJQ0FnZlNrN1hHNGdJQ0FnSUNBZ0lDOHFLaW9xS2lvdklIWmhjaUJmWDNkbFluQmhZMnRmWlhod2IzSjBjMTlmSUQwZ1gxOTNaV0p3WVdOclgzSmxjWFZwY21WZlh5NVBLQ2s3WEc0Z0lDQWdJQ0FnSUM4cUtpb3FLaW92SUY5T1gwVWdQU0JmWDNkbFluQmhZMnRmWlhod2IzSjBjMTlmTzF4dUlDQWdJQ0FnSUNBdktpb3FLaW9xTDF4dUlDQWdJSDBzWEc1ZEtUdGNiaUlLSUNCZExBb2dJQ0oyWlhKemFXOXVJam9nTXdwOUNnPT0zNDMzAHsKICAibWFwcGluZ3MiOiAiOzs7S0FDUyIsCiAgIm5hbWVzIjogWwogICAgInNlbGYiLAogICAgInB1c2giLAogICAgIl9fdW51c2VkX3dlYnBhY2tfbW9kdWxlIiwKICAgICJfX3dlYnBhY2tfZXhwb3J0c19fIiwKICAgICJfX3dlYnBhY2tfcmVxdWlyZV9fIiwKICAgICJTdGF0aWNQYWdlIiwKICAgICJfcmVmIiwKICAgICJkYXRhIiwKICAgICJyZWFjdF9qc3hfcnVudGltZV9fV0VCUEFDS19JTVBPUlRFRF9NT0RVTEVfMF9fIiwKICAgICJqc3giLAogICAgImNoaWxkcmVuIiwKICAgICJmb28iLAogICAgInIiLAogICAgImQiLAogICAgIl9fTl9TU0ciLAogICAgImRlZmF1bHQiLAogICAgIl9fdW51c2VkX3dlYnBhY2tfZXhwb3J0cyIsCiAgICAid2luZG93IiwKICAgICJfX05FWFRfUCIsCiAgICAiTyIsCiAgICAicyIsCiAgICAiX05fRSIKICBdLAogICJzb3VyY2VzIjogWwogICAgIndlYnBhY2s6Ly8uL3BhZ2VzL3N0YXRpYy5qcyIsCiAgICAid2VicGFjazovLy4uLy4uLy4uL3BhY2thZ2VzL25leHQvZGlzdC9idWlsZC93ZWJwYWNrL2xvYWRlcnMvbmV4dC1jbGllbnQtcGFnZXMtbG9hZGVyLmpzP3BhZ2U9JTJGc3RhdGljJmFic29sdXRlUGFnZVBhdGg9cHJpdmF0ZS1uZXh0LXBhZ2VzJTJGc3RhdGljLmpzISIsCiAgICAiLi4vLi4vaW5wdXQvaW5kZXguanMiCiAgXSwKICAic291cmNlc0NvbnRlbnQiOiBbCiAgICAiZXhwb3J0IGRlZmF1bHQgZnVuY3Rpb24gU3RhdGljUGFnZSh7IGRhdGEgfSkge1xuICByZXR1cm4gPGRpdj57ZGF0YS5mb299PC9kaXY+XG59XG5cbmV4cG9ydCBhc3luYyBmdW5jdGlvbiBnZXRTdGF0aWNQcm9wcygpIHtcbiAgcmV0dXJuIHtcbiAgICBwcm9wczoge1xuICAgICAgZGF0YToge1xuICAgICAgICBmb286ICdiYXInLFxuICAgICAgfSxcbiAgICB9LFxuICB9XG59XG4iLAogICAgIlxuICAgICh3aW5kb3cuX19ORVhUX1AgPSB3aW5kb3cuX19ORVhUX1AgfHwgW10pLnB1c2goW1xuICAgICAgXCIvc3RhdGljXCIsXG4gICAgICBmdW5jdGlvbiAoKSB7XG4gICAgICAgIHJldHVybiByZXF1aXJlKFwicHJpdmF0ZS1uZXh0LXBhZ2VzL3N0YXRpYy5qc1wiKTtcbiAgICAgIH1cbiAgICBdKTtcbiAgICBpZihtb2R1bGUuaG90KSB7XG4gICAgICBtb2R1bGUuaG90LmRpc3Bvc2UoZnVuY3Rpb24gKCkge1xuICAgICAgICB3aW5kb3cuX19ORVhUX1AucHVzaChbXCIvc3RhdGljXCJdKVxuICAgICAgfSk7XG4gICAgfVxuICAiLAogICAgIihzZWxmW1wid2VicGFja0NodW5rX05fRVwiXSA9IHNlbGZbXCJ3ZWJwYWNrQ2h1bmtfTl9FXCJdIHx8IFtdKS5wdXNoKFtcbiAgICBbMTU4XSxcbiAgICB7XG4gICAgICAgIC8qKiovIDI5NDM6IC8qKiovIGZ1bmN0aW9uIChcbiAgICAgICAgICAgIF9fdW51c2VkX3dlYnBhY2tfbW9kdWxlLFxuICAgICAgICAgICAgX193ZWJwYWNrX2V4cG9ydHNfXyxcbiAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX19cbiAgICAgICAgKSB7XG4gICAgICAgICAgICBcInVzZSBzdHJpY3RcIjtcbiAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX18ucihfX3dlYnBhY2tfZXhwb3J0c19fKTtcbiAgICAgICAgICAgIC8qIGhhcm1vbnkgZXhwb3J0ICovIF9fd2VicGFja19yZXF1aXJlX18uZChfX3dlYnBhY2tfZXhwb3J0c19fLCB7XG4gICAgICAgICAgICAgICAgLyogaGFybW9ueSBleHBvcnQgKi8gX19OX1NTRzogZnVuY3Rpb24gKCkge1xuICAgICAgICAgICAgICAgICAgICByZXR1cm4gLyogYmluZGluZyAqLyBfX05fU1NHO1xuICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICAgICAgLyogaGFybW9ueSBleHBvcnQgKi8gZGVmYXVsdDogZnVuY3Rpb24gKCkge1xuICAgICAgICAgICAgICAgICAgICByZXR1cm4gLyogYmluZGluZyAqLyBTdGF0aWNQYWdlO1xuICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICAgICAgLyogaGFybW9ueSBleHBvcnQgKi9cbiAgICAgICAgICAgIH0pO1xuICAgICAgICAgICAgLyogaGFybW9ueSBpbXBvcnQgKi8gdmFyIHJlYWN0X2pzeF9ydW50aW1lX19XRUJQQUNLX0lNUE9SVEVEX01PRFVMRV8wX18gPVxuICAgICAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX18oNDUxMik7XG5cbiAgICAgICAgICAgIHZhciBfX05fU1NHID0gdHJ1ZTtcbiAgICAgICAgICAgIGZ1bmN0aW9uIFN0YXRpY1BhZ2UoX3JlZikge1xuICAgICAgICAgICAgICAgIHZhciBkYXRhID0gX3JlZi5kYXRhO1xuICAgICAgICAgICAgICAgIHJldHVybiAvKiNfX1BVUkVfXyovICgwLFxuICAgICAgICAgICAgICAgIHJlYWN0X2pzeF9ydW50aW1lX19XRUJQQUNLX0lNUE9SVEVEX01PRFVMRV8wX18uanN4KShcImRpdlwiLCB7XG4gICAgICAgICAgICAgICAgICAgIGNoaWxkcmVuOiBkYXRhLmZvbyxcbiAgICAgICAgICAgICAgICB9KTtcbiAgICAgICAgICAgIH1cblxuICAgICAgICAgICAgLyoqKi9cbiAgICAgICAgfSxcblxuICAgICAgICAvKioqLyA3MTM5OiAvKioqLyBmdW5jdGlvbiAoXG4gICAgICAgICAgICBfX3VudXNlZF93ZWJwYWNrX21vZHVsZSxcbiAgICAgICAgICAgIF9fdW51c2VkX3dlYnBhY2tfZXhwb3J0cyxcbiAgICAgICAgICAgIF9fd2VicGFja19yZXF1aXJlX19cbiAgICAgICAgKSB7XG4gICAgICAgICAgICAod2luZG93Ll9fTkVYVF9QID0gd2luZG93Ll9fTkVYVF9QIHx8IFtdKS5wdXNoKFtcbiAgICAgICAgICAgICAgICBcIi9zdGF0aWNcIixcbiAgICAgICAgICAgICAgICBmdW5jdGlvbiAoKSB7XG4gICAgICAgICAgICAgICAgICAgIHJldHVybiBfX3dlYnBhY2tfcmVxdWlyZV9fKDI5NDMpO1xuICAgICAgICAgICAgICAgIH0sXG4gICAgICAgICAgICBdKTtcbiAgICAgICAgICAgIGlmIChmYWxzZSkge1xuICAgICAgICAgICAgfVxuXG4gICAgICAgICAgICAvKioqL1xuICAgICAgICB9LFxuICAgIH0sXG4gICAgLyoqKioqKi8gZnVuY3Rpb24gKF9fd2VicGFja19yZXF1aXJlX18pIHtcbiAgICAgICAgLy8gd2VicGFja1J1bnRpbWVNb2R1bGVzXG4gICAgICAgIC8qKioqKiovIHZhciBfX3dlYnBhY2tfZXhlY19fID0gZnVuY3Rpb24gKG1vZHVsZUlkKSB7XG4gICAgICAgICAgICByZXR1cm4gX193ZWJwYWNrX3JlcXVpcmVfXygoX193ZWJwYWNrX3JlcXVpcmVfXy5zID0gbW9kdWxlSWQpKTtcbiAgICAgICAgfTtcbiAgICAgICAgLyoqKioqKi8gX193ZWJwYWNrX3JlcXVpcmVfXy5PKDAsIFs3NzQsIDg4OCwgMTc5XSwgZnVuY3Rpb24gKCkge1xuICAgICAgICAgICAgcmV0dXJuIF9fd2VicGFja19leGVjX18oNzEzOSk7XG4gICAgICAgIH0pO1xuICAgICAgICAvKioqKioqLyB2YXIgX193ZWJwYWNrX2V4cG9ydHNfXyA9IF9fd2VicGFja19yZXF1aXJlX18uTygpO1xuICAgICAgICAvKioqKioqLyBfTl9FID0gX193ZWJwYWNrX2V4cG9ydHNfXztcbiAgICAgICAgLyoqKioqKi9cbiAgICB9LFxuXSk7XG4iCiAgXSwKICAidmVyc2lvbiI6IDMKfQo=)